core/src/main/java/org/apache/druid/java/util/metrics/AllocationMetricCollectorFactory.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.java.util.metrics;
+
+import org.apache.druid.java.util.common.logger.Logger;
+
+import java.lang.management.ManagementFactory;
+import java.lang.management.ThreadMXBean;
+import java.lang.reflect.Method;
+
+class AllocationMetricCollectorFactory
```

Comment:
Please add private constructor to prohibit accidental instantiation

Reply:
This class name is confusing for me because if a class is called "XxxFactory" I expect that I could instantiate it and get an "xxx" by calling an instance method of the factory. Static factory method containers are usually called "Xxxs", i. e. `AllocationMetricCollectors` in this case.

Reply:
You've got the instance method `getAllocationMetricCollector`.  I didn't get it.